code/AssetLib/glTF2/glTF2Importer.cpp

```diff
@@ -1640,7 +1640,7 @@ void glTF2Importer::ImportEmbeddedTextures(glTF2::Asset &r) {
 
                 const size_t len = strlen(ext);
                 if (len <= 3) {
-                    strncpy(tex->achFormatHint, ext, len);
+                    strncpy(tex->achFormatHint, ext, len+1);//+1 because we need to copy terminator
```

Comment:
This will cause an override when the length is more than 4 bytes. Just using a fixed-sized array would work better. The default size is 4.

Reply:
isn't the check len<=3 on top guarantees it won't be bigger than 4? That was what I though at least.

Reply:
I guess the terminator will be added automatically.

Reply:
+1 is to copy the terminator. strlen function gives you the lenght of the string without the termination character. if you copy only the bytes of the strlen(ext) result, you are not copying the terminator. you have to copy 1 more bytes to copy the \0. 

texture.h shows "tex->achFormatHint" value is memset to 0, so this won't matter unless we are rewriting it from a longer hint.